kernel/sched.c

```diff
@@ -1065,17 +1068,19 @@ void z_sched_abort(struct k_thread *thread)
 	 * running on or because we caught it idle in the queue
 	 */
 	while ((thread->base.thread_state & _THREAD_DEAD) == 0U) {
-		LOCKED(&sched_spinlock) {
-			if (z_is_thread_prevented_from_running(thread)) {
-				__ASSERT(!z_is_thread_queued(thread), "");
-				thread->base.thread_state |= _THREAD_DEAD;
-			} else if (z_is_thread_queued(thread)) {
-				_priq_run_remove(&_kernel.ready_q.runq, thread);
-				z_mark_thread_as_not_queued(thread);
-				thread->base.thread_state |= _THREAD_DEAD;
-			} else {
-				k_busy_wait(100);
-			}
+		key = k_spin_lock(&sched_spinlock);
+		if (z_is_thread_prevented_from_running(thread)) {
+			__ASSERT(!z_is_thread_queued(thread), "");
+			thread->base.thread_state |= _THREAD_DEAD;
+			k_spin_unlock(&sched_spinlock, key);
+		} else if (z_is_thread_queued(thread)) {
+			_priq_run_remove(&_kernel.ready_q.runq, thread);
+			z_mark_thread_as_not_queued(thread);
+			thread->base.thread_state |= _THREAD_DEAD;
+			k_spin_unlock(&sched_spinlock, key);
+		} else {
+			k_spin_unlock(&sched_spinlock, key);
+			k_busy_wait(100);
```

Comment:
There's another patch out there with exactly this change.  I agree with the logic, but have the same style comment here: can you leave the outer lock/unlock pair at the same level of indentation. and just put an unlock/lock around the k_busy_wait() call itself?

IMHO lock maintenance gets hard, and needs to be expressed in easily seen matched pairs wherever possible.

Reply:
after #20046 is merged,  i will make a clean up of this PR.